File: app/src/main/java/com/sp/base/database/AppDatabase.java
package com.sp.base.database;

import androidx.room.Database;
import androidx.room.RoomDatabase;
import androidx.room.TypeConverters;
@Database(entities = {UserProfileDetailsModel.class}, version = 1, exportSchema = false)
@TypeConverters({JsonTypeConverter.class})
public abstract class AppDatabase extends RoomDatabase {
    public abstract DBDAO getDbDAO();
}

File: app/src/main/java/com/baseapplication/database/AppDatabase.kt
package com.baseapplication.database

import androidx.room.Database
import androidx.room.RoomDatabase
import androidx.room.TypeConverters

@Database(entities = [UserProfileDetailsModel::class], version = 1, exportSchema = false)
@TypeConverters(*[JsonTypeConverter::class])
abstract class AppDatabase() : RoomDatabase() {
    abstract val dbDAO: DBDAO
}
